drivers/node/aws/aws.go

```diff
@@ -0,0 +1,128 @@
+package aws
+
+import (
+	"fmt"
+	"github.com/portworx/torpedo/drivers/node"
+	//"github.com/portworx/torpedo/drivers/scheduler"
+	aws_pkg "github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"log"
+)
+
+const (
+	// DriverName is the name of the aws driver
+	DriverName = "aws"
+	// AwsRegion is the region of aws ec2
+	AwsRegion = "us-east-1"
+)
+
+type aws struct {
+	node.Driver
+	session     *session.Session
+	credentials *credentials.Credentials
+	config      *aws_pkg.Config
+	svc         *ec2.EC2
+	Instances   []*ec2.Instance
+}
+
+func (aws *aws) String() string {
+	return DriverName
+}
+
+func (aws *aws) Init(sched string) error {
+	sess := session.Must(session.NewSession())
+	aws.session = sess
+	creds := credentials.NewEnvCredentials()
+	aws.credentials = creds
+	config := &aws_pkg.Config{Region: aws_pkg.String(AwsRegion)}
+	config.WithCredentials(creds)
+	aws.config = config
+	svc := ec2.New(sess, config)
+	aws.svc = svc
+	//Get all nodes from east region
+	instances := aws.getAllInstancesInEast()
+	aws.Instances = instances
+	return nil
+}
+
+func (aws *aws) TestConnection(n node.Node, options node.TestConectionOpts) error {
+	return nil
+}
+
+func (aws *aws) RebootNode(n node.Node, options node.RebootNodeOpts) error {
+	var err error
+	_, err = aws.getAddrToConnect(n)
+	if err != nil {
+		return &ErrFailedToRebootNode{
+			Node:  n,
+			Cause: fmt.Sprintf("failed to get node address due to: %v", err),
+		}
+	}
+	instanceID, err := aws.getNodeIDByPrivAddr(n)
+	if err != nil {
+		fmt.Println("there was an error get instance ID by its private IP ", n.Addresses[0], err.Error())
+		log.Fatal(err.Error())
+	}
+	//Reboot the instance by its InstanceID
+	rebootInput := &ec2.RebootInstancesInput{
+		InstanceIds: []*string{
+			aws_pkg.String(instanceID),
+		},
+	}
+	fmt.Printf("Now reboot instance %+v\n", rebootInput)
+	_, err = aws.svc.RebootInstances(rebootInput)
+	if err != nil {
+		return fmt.Errorf("there was an error reboot instances %s, %q", instanceID, err.Error())
+	}
+	return nil
+}
+
+func (aws *aws) ShutdownNode(n node.Node, options node.ShutdownNodeOpts) error {
+	return nil
+}
+
+func (aws *aws) getAddrToConnect(n node.Node) (string, error) {
+	if n.Addresses == nil || len(n.Addresses) == 0 {
+		return "", fmt.Errorf("no address available to connect")
+	}
+	addr := n.Addresses[0] // TODO don't stick to first address
+	return addr, nil
+}
+
+func (aws *aws) getAllInstancesInEast() []*ec2.Instance {
```

Comment:
Why does this function name say East? Within the function, there is nothing specific about the east region.

In general, the code needs to be generic for aws and variables like regions should be provided during init.

Reply:
Changed to getAllInstances()